src/scripts.rs

```diff
@@ -361,6 +361,59 @@ fn unvault_descriptor_csv<Pk: MiniscriptKey>(desc: &Descriptor<Pk>) -> u32 {
     }
 }
 
+fn unvault_descriptor_managers_threshold<Pk: MiniscriptKey>(
+    desc: &Descriptor<Pk>,
+    manager_keys: Vec<Pk>,
+) -> usize {
+    let ms = match &desc {
+        Descriptor::Wsh(ref wsh) => match wsh.as_inner() {
+            WshInner::Ms(ms) => ms,
+            WshInner::SortedMulti(_) => unreachable!("Unvault descriptor is not a sorted multi"),
+        },
+        _ => unreachable!("Unvault descriptor is always a P2WSH"),
+    };
+
+    let policy = ms.lift().unwrap();
+
+    rec_unvault_descriptor_managers_threshold(
+        &policy,
+        &manager_keys
+            .into_iter()
+            .map(|k| k.to_pubkeyhash())
+            .collect::<Vec<_>>(),
+    )
+}
+
+fn rec_unvault_descriptor_managers_threshold<Pk: MiniscriptKey>(
+    policy: &SemanticPolicy<Pk>,
+    manager_keys: &[Pk::Hash],
+) -> usize {
+    // Looks for the manager threshold: we recursively explore the policy tree, returning the threshold
+    // if we find it, 0 otherwise. We sum our results at each step and tada!
+    // Note that we would return the wrong result if, for some reason, we had two different
+    // branches with all the manager keys. But we know the structure of the unvault_descriptor
+    // and we know for sure we would find the managers in only one branch, so all good!
```

Comment:
If we are sure about that we should panic rather than returning an invalid value. We would never create a descriptor with `0` as thresh.